dask/dot.py

```diff
@@ -306,3 +308,237 @@ def graphviz_to_file(g, filename, format):
         f.write(data)
 
     return display_cls(filename=full_filename)
+
+
+def to_cytoscape_json(
+    dsk,
+    data_attributes=None,
+    function_attributes=None,
+    collapse_outputs=False,
+    verbose=False,
+    **kwargs,
+):
+    """
+    Convert a dask graph to Cytoscape JSON:
+    https://js.cytoscape.org/#notation/elements-json
+    """
+    nodes = []
+    edges = []
+    data = {"nodes": nodes, "edges": edges}
+
+    data_attributes = data_attributes or {}
+    function_attributes = function_attributes or {}
+
+    seen = set()
+    connected = set()
+
+    for k, v in dsk.items():
```

Comment:
It's unfortunate we have to largely repeat this logic. How difficult would it be to centralize this logic and use it in both `to_graphviz` and here?

Reply:
I thought about this, but there are enough differences between names of various properties that it seemed premature to me (I'd want to either have some sort of wrapper protocol or a raft of `if` statements). I might think differently if there were a third visualizer, though.